Packet++/header/DoIpLayer.h

```diff
@@ -0,0 +1,1802 @@
+
+		switch (payloadType)
+		{
+		case DoIpPayloadTypes::ALIVE_CHECK_REQUEST:
+		case DoIpPayloadTypes::ALIVE_CHECK_RESPONSE:
+		case DoIpPayloadTypes::ANNOUNCEMENT_MESSAGE:
+		case DoIpPayloadTypes::DIAGNOSTIC_MESSAGE_NEG_ACK:
+		case DoIpPayloadTypes::DIAGNOSTIC_MESSAGE_POS_ACK:
+		case DoIpPayloadTypes::DIAGNOSTIC_MESSAGE_TYPE:
+		case DoIpPayloadTypes::DIAGNOSTIC_POWER_MODE_REQUEST:
+		case DoIpPayloadTypes::DIAGNOSTIC_POWER_MODE_RESPONSE:
+		case DoIpPayloadTypes::ENTITY_STATUS_REQUEST:
+		case DoIpPayloadTypes::ENTITY_STATUS_RESPONSE:
+		case DoIpPayloadTypes::GENERIC_HEADER_NEG_ACK:
+		case DoIpPayloadTypes::ROUTING_ACTIVATION_REQUEST:
+		case DoIpPayloadTypes::ROUTING_ACTIVATION_RESPONSE:
+		case DoIpPayloadTypes::VEHICLE_IDENTIFICATION_REQUEST:
+		case DoIpPayloadTypes::VEHICLE_IDENTIFICATION_REQUEST_WITH_EID:
+		case DoIpPayloadTypes::VEHICLE_IDENTIFICATION_REQUEST_WITH_VIN:
+			return true;
+
+		default:
+			PCPP_LOG_DEBUG("[Malformed doip packet]: Invalid DoIP payload type: 0x" << std::hex << type);
+			return false;
+		}
+	}
+
+	inline bool DoIpLayer::isPayloadLengthValid(uint32_t payloadLength, size_t dataLen)
+	{
+		const size_t actualPayloadLen = dataLen - DOIP_HEADER_LEN;
+
+		if (payloadLength != actualPayloadLen)
+		{
+			PCPP_LOG_DEBUG("[Malformed doip packet]: Payload length mismatch: expected "
+			               << payloadLength << " bytes, but got " << actualPayloadLen << " bytes.");
+			return false;
+		}
+
+		return true;
+	}
+
+	//~~~~~~~~~~~~~~~~~~~~~~~~~|
+	// RoutingActivationRequest|
+	//~~~~~~~~~~~~~~~~~~~~~~~~~|
+
+	/// @class DoIpRoutingActivationRequest
+	/// @brief Represents a DoIP Routing Activation Request message.
+	///
+	/// Provides parsing and construction for Routing Activation Request messages
+	/// as defined by the DoIP protocol.
+	class DoIpRoutingActivationRequest : public DoIpLayer
+	{
+	public:
+		/// @brief Constructs the layer from raw DoIP packet data.
+		/// @param[in] data Pointer to raw packet data.
+		/// @param[in] dataLen Length of the raw data.
+		/// @param[in] prevLayer Pointer to the previous layer.
+		/// @param[in] packet Pointer to the parent packet instance.
+		DoIpRoutingActivationRequest(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet);
+
+		/// @brief Constructs the message from field values.
+		/// @param[in] sourceAddress Source address of the tester.
+		/// @param[in] activationType Type of routing activation.
+		DoIpRoutingActivationRequest(uint16_t sourceAddress, DoIpActivationTypes activationType);
+
+/// @struct  routing_activation_request
+/// Represents the Routing Activation Request data in DoIP.
+/// Routing Activation Response message structure (extends DoIP header).
+#pragma pack(push, 1)
+		struct routing_activation_request : doiphdr
+		{
+			/// @brief Source address of the tester.
+			uint16_t sourceAddress;
+
+			/// @brief Routing activation type.
+			DoIpActivationTypes activationType;
```

Comment:
This struct contains an enum class - `DoIpActivationTypes`, so if we do a `reinterpret_cast<routing_activation_request*>(data)` and the activation type contains an invalid value, the behavior is undefined. I'd keep a `uint8_t` value in `routing_activation_request` and do this conversion in the getter `DoIpRoutingActivationRequest::getActivationType()` which can handle invalid values.

Same for the other structs below - they should all contain `uint8_t` or `uint16_t` instead of enum classes